backend/functions/lib/adapters/gateways/azure-pacer-secret.gateway.test.ts

```diff
@@ -0,0 +1,92 @@
+import { AzurePacerSecretsGateway } from './azure-pacer-secrets.gateway';
+import { NoPacerToken } from './pacer-exceptions';
+import { AzureKeyVaultGateway } from './azure-key-vault.gateway';
+const context = require('../../testing/defaultContext');
+
+class MockSecretsGateway extends AzureKeyVaultGateway {
+
+  constructor(functions?: { setSecret?: any; getSecret?: any }) {
+    super();
+    if (functions?.setSecret) {
+      this.setSecret = functions.setSecret;
+    }
+    if (functions?.getSecret) {
+      this.getSecret = functions.getSecret;
+    }
+  }
+
+  setSecret(): Promise<string> {
+    return Promise.resolve('fake-new-token');
+  }
+
+  async getSecret(name: string): Promise<string> {
+    return Promise.resolve('fake-token');
+  }
+}
+
+describe('Azure PACER secrets gateway test', () => {
+
+  beforeAll(() => {
+    process.env = {
+      PACER_TOKEN_SECRET_NAME: 'fake-secret',
+    };
+  });
+
+  test('should throw error', async () => {
+    const gateway = new AzurePacerSecretsGateway(new MockSecretsGateway(
+      {
+        setSecret: async () => {
+          throw new Error('some error');
+        }
+      }
+    ));
+
+    try {
+      await gateway.savePacerTokenToSecrets('fake-new-token', context);
+      expect(true).toBeFalsy();
+    } catch (e) {
+      expect(e.message).toEqual('some error');
+    }
+  });
+
+  test('should return PACER token', async () => {
+    const gateway = new AzurePacerSecretsGateway(new MockSecretsGateway());
+
+    const token = await gateway.getPacerTokenFromSecrets(context);
+    expect(token).toEqual('fake-token');
+  });
+
+  test('should throw NoPacerToken error', async () => {
+    const gateway = new AzurePacerSecretsGateway(new MockSecretsGateway(
+      {
+        getSecret: async () => {
+          throw new Error(`The secret fake-secret was not found.`);
+        }
+      }
+    ));
+
+    try {
+      await gateway.getPacerTokenFromSecrets(context);
+      expect(true).toBeFalsy();
+    } catch (e) {
+      expect(e).toBeInstanceOf(NoPacerToken);
+    }
+  });
+
+  test('should throw  error', async () => {
```

Comment:
// remove the extra space.
```suggestion
  test('should throw error', async () => {
```